lib/ASTGen/Sources/ASTGen/PluginHost.swift

```diff
@@ -378,6 +374,26 @@ class PluginDiagnosticsEngine {
   }
 }
 
+extension String {
+  /// Retrieve the base name of a string that represents a path, removing the
+  /// directory.
+  var basename: String {
+    guard
+      let lastSlash = lastIndex(where: {
+        #if os(iOS) || os(macOS) || os(tvOS) || os(watchOS) || os(visionOS) || os(Android) || os(Linux)
+        ["/"].contains($0)
+        #else
+        ["/", "\\"].contains($0)
+        #endif
+      })
+    else {
+      return self
+    }
+
+    return String(self[index(after: lastSlash)...])
+  }
```

Comment:
I don’t think this is used anywhere.

Reply:
This is moved from deleted [lib/ASTGen/Sources/ASTGen/SourceManager+MacroExpansionContext.swift](https://github.com/apple/swift/pull/73725#diff-837202e3d3c3d5f4d1db10aff9aebdabdae57a6d16e58fc184c06266eb542844)
And is used at https://github.com/apple/swift/blob/03b0f0cc57feb18246348168fb2f937e95496277/lib/ASTGen/Sources/ASTGen/PluginHost.swift#L394-L397

Reply:
Ah, that’s why I didn’t see it.